### DISCUSSION TITLE
Stuck on the issue referenced in: Discussion #50 (Row-based layout for FormCollection). Utilizing FieldsetMixin with ModelForm fails to render properly

### DISCUSSION THREAD

**Discussion by omarcoming** [jrief/django-formset#69] [Q&A]
I would like to use Fieldsets with my collection but none of the css attributes I pass get rendered.  My problem is most similar to Discussion #50 but the matter was resolved as jwoithe did not need to use Fieldset's features. 

EDIT:  Sorry please ignore everything below.  The solutions below do implement the CSS classes (although I can't seem to figure out how to apply css classes to the entire collection instead of each subcollection). However this is still only if I'm not utililzing FieldsetMixin so that problem still remains. 

--------------------------------------------------------------------------------------------------------------------------------------------------------

_I've tried both of [these methods in Discussion #43](https://github.com/jrief/django-formset/discussions/43#discussioncomment-5352250): 

As well as the solution found by gumish [here](https://github.com/jrief/django-formset/discussions/43#discussioncomment-5357963): 

All to no avail. If I do not inherit from FieldsetMixin, I can get just the fields to render by setting default_renderer in the form or upon instantiating the form but neither setting the renderer in the collection nor upon instantiation of the collection has any effect. 
 
The following renders the form and field but the hide_if doesn't work. (as does setting the default_renderer in CustomerForm) 
```
class CustomerForm(ModelForm):
    legend = Customer
    hide_if = 'ext_customer.extant'

    class Meta:
        model = Customer
        fields = [
            'first_name',
        ]
        widgets = {
            'first_name': forms.TextInput(attrs={
                'class': 'form-control',
                'id': 'first_name',
                'placeholder': 'Customer First Name',
            })
        }


class ExtantCustomerForm(forms.Form):
    extant = fields.BooleanField(
        label='Existing Customer?',
        required=False,
    )


class CustomerCollection(FormCollection):
    customer = CustomerForm(renderer=FormRenderer(
        form_css_classes='row',
        fieldset_css_classes='row border',
        field_css_classes={
            '*': 'mb-1 col-4',
            'city': 'mb-1 col-3',
            'state': 'mb-1 col-1',
            'customer_notes': 'mb-5 col-12',
        },
        collection_css_classes='row border'
    ))

class InvoiceCollection(FormCollection):
    default_renderer = FormRenderer()
    customer = CustomerCollection()
```

Setting renderer in CustomerCollection or setting render= when instantiating CustomerCollection in InvoiceCollection fail to pass _any_ of the css classes._

**Comment by jrief** [comment]
Bootstrap-5 sets the border of `fieldset` to none. In order to overwrite this setting, please add `fieldset_css_classes="border rounded p-3"` to the form renderer used to render that fieldset.

**Reply by omarcoming** [reply]
Is FieldsetMixin incompatible with ModelForm?

**Reply by jrief** [reply]
It shouldn't. I have to test this though.
